src/musig_hash.rs

```diff
@@ -0,0 +1,86 @@
+use bellman::pairing::ff::{PrimeField, PrimeFieldRepr};
+use franklin_crypto::eddsa::PublicKey;
+use franklin_crypto::jubjub::JubjubEngine;
+use franklin_crypto::util::sha256_hash_to_scalar;
+use sha2::{Digest, Sha256};
+
+pub trait AggregateHash<E: JubjubEngine> {
+    fn hash(&self, pubs: &[PublicKey<E>], last: &PublicKey<E>) -> E::Fs;
```

Comment:
```
Is it possible that one will want to use AggregateHash but totally not interested in CommitmentHash in the context of this library?
```
It's not, but it doesn't mean we should combine all of the traits into one. Because:
1. It break SOLID principles
2. It would require us to create adapters for every combination of hashes we want to use (e.g. using musig with SHA512 + rescue and SHA512 + pedersen will require to create two structs on calling side with corresponding trait implementation)
3. This was a try to fix existing Franklin crypto problem with creating separate copy&paste functions for musig (e.g. musig_sha256_sign, musig_pedersen_sign, musig_rescue_sign) which would become even worse in case of using several hashes per algorithm like in this case.